omplConfig.cmake.in

```diff
@@ -18,22 +19,45 @@ set(OMPL_MINOR_VERSION @PROJECT_VERSION_MINOR@)
 set(OMPL_PATCH_VERSION @PROJECT_VERSION_PATCH@)
 
 set_and_check(OMPL_INCLUDE_DIR "@PACKAGE_INCLUDE_INSTALL_DIR@")
-set(OMPL_INCLUDE_DIRS "${OMPL_INCLUDE_DIR};@EIGEN3_INCLUDE_DIR@;@FCL_INCLUDE_DIRS@;@ASSIMP_INCLUDE_DIRS@")
+set(OMPL_INCLUDE_DIRS "${OMPL_INCLUDE_DIR};@Boost_INCLUDE_DIR@;@EIGEN3_INCLUDE_DIR@")
+foreach(_dir "@FLANN_INCLUDE_DIRS@;@ODE_INCLUDE_DIRS@;@SPOT_INCLUDE_DIRS@;@TRIANGLE_INCLUDE_DIR@;@FCL_INCLUDE_DIRS@;@PQP_INCLUDE_DIR@;@ASSIMP_INCLUDE_DIRS@;@OPENGL_INCLUDE_DIR@")
+    if(_dir)
+        list(APPEND OMPL_INCLUDE_DIRS "${_dir})
+    endif()
+endforeach()
 list(REMOVE_DUPLICATES OMPL_INCLUDE_DIRS)
-list(REMOVE_ITEM OMPL_INCLUDE_DIRS "")
 set(OMPL_INCLUDE_DIRS "${OMPL_INCLUDE_DIRS}" CACHE STRING "Include path for OMPL and its dependencies")
 
 set_and_check(OMPL_LIBRARY_DIR @PACKAGE_LIB_INSTALL_DIR@)
+set(OMPL_LIBRARY_DIRS "${OMPL_LIBRARY_DIR};@Boost_LIBRARY_DIRS@")
+foreach(_dir "@FLANN_LIBRARY_DIRS@;@ODE_LIBRARY_DIRS@;@SPOT_LIBRARY_DIRS@;@FCL_LIBRARY_DIRS@;@ASSIMP_LIBRARY_DIRS@")
+    if(_dir)
+        list(APPEND OMPL_LIBRARY_DIRS "${_dir})
+    endif()
+endforeach()
+list(REMOVE_DUPLICATES OMPL_INCLUDE_DIRS)
+set(OMPL_LIBRARY_DIRS "${OMPL_LIBRARY_DIRS}" CACHE STRING "Library path for OMPL and its dependencies")
+
 find_library(OMPL_LIBRARIES NAMES ompl.${OMPL_VERSION} ompl
     PATHS ${OMPL_LIBRARY_DIR} NO_DEFAULT_PATH)
+foreach(_lib "@Boost_SERIALIZATION_LIBRARY@;@Boost_FILESYSTEM_LIBRARY@;@Boost_SYSTEM_LIBRARY@;@ODE_LIBRARIES@;@SPOT_LIBRARIES@")
+    if(_lib)
+        list(APPEND OMPL_LIBRARIES "${_lib}")
+    endif()
+endforeach()
+
 find_library(OMPLAPPBASE_LIBRARY NAMES ompl_app_base.${OMPL_VERSION} ompl_app_base
     PATHS ${OMPL_LIBRARY_DIR} NO_DEFAULT_PATH)
 find_library(OMPLAPP_LIBRARY NAMES ompl_app.${OMPL_VERSION} ompl_app
     PATHS ${OMPL_LIBRARY_DIR} NO_DEFAULT_PATH)
 if (OMPLAPPBASE_LIBRARY AND OMPLAPP_LIBRARY)
-    set(OMPLAPP_LIBRARIES "${OMPLAPPBASE_LIBRARY};${OMPLAPP_LIBRARY}"
+    set(OMPLAPP_LIBRARIES "${OMPLAPPBASE_LIBRARY};${OMPLAPP_LIBRARY};@ASSIMP_LIBRARIES@;@FCL_LIBRARIES@"
+    if(@OPENGL_LIBRARIES@)
+        list(APPEND OMPLAPP_LIBRARIES "${_lib}")
+    endif()
+    set(OMPLAPP_LIBRARIES "${OMPLAPP_LIBRARIES}"
         CACHE STRING "Paths to OMPL.app libraries")
 endif()
 
```

Comment:
It may be good to keep the list remove duplicates for the variables. I am getting repeat include directories